contrib/opentimelineio_contrib/application_plugins/rv/example_otio_writer/otio_writer.py

```diff
@@ -0,0 +1,378 @@
+        return otio.schema.Clip(
+            name=rv.extra_commands.uiName(node_name),
+            media_reference=media_reference,
+            source_range=source_range,
+            metadata=metadata
+        )
+
+
+def _create_transition(node_name, in_frame=None, out_frame=None,
+                       cut_in_frame=None):
+    """
+    Create an OTIO Transition for a CrossDissolve node.
+    :param node_name: `str`
+    :return: `otio.schema.Transition`
+    """
+    transition_type = TRANSITION_TYPE_MAP.get(
+        rv.commands.nodeType(node_name),
+        otio.schema.transition.TransitionTypes.Custom
+    )
+    if not transition_type:
+        return None
+
+    input_node_names, _ = rv.commands.nodeConnections(node_name)
+
+    # Assume FPS matches first input (RV Transition don't have FPS settings)
+    fps = get_source_fps(input_node_names[0])
+
+    durations_frames = rv.commands.getFloatProperty(
+        node_name + ".parameters.numFrames")
+    if not durations_frames:
+        duration_frames = 20.0
+    else:
+        duration_frames = durations_frames[0]
+
+    in_offset_frames = int(math.floor(duration_frames/2))
+    in_offset = otio.opentime.RationalTime(in_offset_frames, rate=fps)
+    out_offset = otio.opentime.RationalTime(duration_frames-in_offset_frames,
+                                            rate=fps)
+
+    transition = otio.schema.Transition(
+        name=rv.extra_commands.uiName(node_name),
+        transition_type=transition_type,
+        in_offset=in_offset,
+        out_offset=out_offset,
+        metadata=get_node_otio_metadata(node_name)
+    )
+    return transition
+
+
+def get_node_otio_metadata(node_name):
+    """
+    Retrieve the value of the OTIO Metadata property for a node, if it exists.
+    This is set by otio_reader.py application plugin.
+    :param node_name: `str`
+    :return: `str`
+    """
+    # Standard OTIO Metadata prop from OTIO Reader
+    tio_prop = node_name + ".attributes.otio.metadata"
+
+    # Persist the Node properties as metadata
+    node_type = rv.commands.nodeType(node_name)
+    metadata = {node_type: {}}
+    for prop in rv.commands.properties(node_name):
+        if prop == otio_prop:
+            continue
+
+        property_info = rv.commands.propertyInfo(prop)
+        if property_info["type"] == rv.commands.IntType:
+            metadata[node_type][prop] = rv.commands.getIntProperty(prop)
+        elif property_info["type"] == rv.commands.FloatType:
+            metadata[node_type][prop] = rv.commands.getFloatProperty(prop)
+        elif property_info["type"] == rv.commands.StringType:
+            metadata[node_type][prop] = rv.commands.getStringProperty(prop)
+
+    # Persist the general OTIO metadata separately to match OTIO Reader
+    if rv.commands.propertyExists(otio_prop):
+        metadata_str = rv.commands.getStringProperty(otio_prop)[0]
+        try:
+            metadata.update(json.loads(metadata_str))
```

Comment:
Hi @JoshBurnell , just thought I would point out some potential changes in #640 (and conversation in #opentimelineio). I think the fix would be as simple as 
```
otio.core.deserialize_json_from_string(metadata_str)
# or.. not sure if it really matters
otio.adapters.read_from_string(metadata_str)
```